libs/langchain/langchain/memory/summary_buffer.py

```diff
@@ -71,6 +72,7 @@ def prune(self) -> None:
             self.moving_summary_buffer = self.predict_new_summary(
                 pruned_memory, self.moving_summary_buffer
             )
+            self.k = len(buffer)
```

Comment:
it pops the buffer above (line 70)

are we sure this is needed?

Reply:
we also need to update k after each iteration if the summary is not pruned

Reply:
okay @hwchase17

Reply:
@hwchase17, why do we need to update 'k' when the summary is not pruned? 
The idea is that the messages which become part of the `moving_summary_buffer` won't be needed in the next loaded buffer. Therefore, 'k' is only updated when the 'moving_summary_buffer' is updated. 
Please see line 37, as I only take the last 'k' messages from underlying memory at that point because the rest of the popped messages have become part of the moving summary. I hope this is clear. Let me know?

Reply:
@hwchase17 sounds good?

Reply:
If yes then I will update the documentation as well.

Reply:
@hwchase17 any updates on this?